src/pages/data.test.js

```diff
@@ -0,0 +1,427 @@
+        )
+
+        const headerBar = screen.getByTestId('dhis2-ui-selectorbar')
+        within(headerBar)
+            .queryByRole('button', { name: 'Clear selections' })
+            .click()
+        expect(screen.getByTestId('data-exchange-selector')).toHaveTextContent(
+            'Choose a data exchange'
+        )
+    })
+
+    it('should display the correct exchange specified in url if the param is present', async () => {
+        const anExchange = testDataExchange()
+        const exchanges = [anExchange, testDataExchange()]
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+        })
+
+        expect(screen.getByTestId('data-exchange-selector')).toHaveTextContent(
+            anExchange.displayName
+        )
+    })
+
+    it('should display a preview table once an exchange id selected', async () => {
+        const anExchange = testDataExchange()
+
+        const exchanges = [anExchange]
+        const exchangeData = testDataExchangeSourceData()
+        const { screen } = setUp(<DataPage />, {
+            aggregateDataExchanges: exchanges,
+            exchangeId: anExchange.id,
+            exchangeData,
+        })
+
+        const titleBar = await screen.findByTestId('title-bar')
+        expect(titleBar).toHaveTextContent(anExchange.displayName)
+        expect(titleBar).toHaveTextContent(
+            `${anExchange.source.requests.length} data report`
+        )
+        const timeDifference = getRelativeTimeDifference({
+            startTimestamp: lastAnalyticsTableSuccess,
+            endTimestamp: serverDate,
+        })
+        expect(titleBar).toHaveTextContent(
+            `Source data was generated ${timeDifference} ago`
+        )
+
+        const tabBar = await screen.findByTestId('dhis2-uicore-tabbar')
+        expect(within(tabBar).getAllByTestId('dhis2-uicore-tab')).toHaveLength(
+            anExchange.source.requests.length
+        )
+        anExchange.source.requests.map((request) => {
+            expect(tabBar).toHaveTextContent(request.name)
+        })
+
+        const dataTable = await screen.findByTestId('dhis2-uicore-datatable')
+        const formattedData = formatData(exchangeData)[0]
+        expect(dataTable).toBeInTheDocument()
+        expect(
+            within(dataTable).getByTestId('dhis2-uicore-tablehead')
+        ).toHaveTextContent(formattedData.title)
+        expect(
+            within(dataTable).getByTestId('dhis2-uicore-tablehead')
+        ).toHaveTextContent(formattedData.title)
+        const tableRows = within(dataTable).getAllByTestId(
+            'dhis2-uicore-datatablerow'
+        )
+        expect(tableRows).toHaveLength(3)
+        formattedData.headers.map((tableHeader) =>
+            expect(tableRows[0]).toHaveTextContent(tableHeader.name)
+        )
+        formattedData.rows[0].map((firstRowCell) =>
+            expect(tableRows[1]).toHaveTextContent(firstRowCell)
```

Comment:
I guess you could have a the formattedData have data like [1,2,3] and this test would pass if the data displayed in the row as [3,1,2]? I don't know if it's worth it to try to make sure that the positioning within the row is correct?